src/Libraries/Microsoft.Extensions.Http.Resilience/Routing/Internal/RoutingStrategy.cs

```diff
@@ -41,16 +34,19 @@ public RoutingPolicy(string pipelineName, IRequestRoutingStrategyFactory factory
             Throw.InvalidOperationException("The routing strategy did not provide any route URL on the first attempt.");
         }
 
-        _routingStrategySetter(context, strategy);
+        if (!context.Properties.TryGetValue(ResilienceKeys.RequestMessage, out var request))
+        {
+            Throw.InvalidOperationException("The request message was not found in the resilience context.");
```

Comment:
Maybe mention it's HTTP?
```suggestion
            Throw.InvalidOperationException("The HTTP request message was not found in the resilience context.");
```